build.gradle

```diff
@@ -41,8 +41,7 @@ allprojects {
 }
 
 dependencies {
-    testImplementation 'org.codehaus.groovy:groovy-all:2.4.1'
-    testImplementation 'org.spockframework:spock-core:1.0-groovy-2.4'
+    testImplementation 'org.spockframework:spock-core:1.3-groovy-2.5'
```

Comment:
Spock 1.3 requires Java 8. That means that the tests can't run on anything < Java 8. Since we seem to only manually test for anything less than Java 8 that could be fine.

Reply:
Looks like I misread the sentence. Spock 1.2 and 1.3 require Java 7 (they drop support for Java 6): http://spockframework.org/spock/docs/1.3/release_notes.html#_1_2_2018_09_23

Still something to consider.

Reply:
@adammurdoch can we drop automated testing for Java 5 and 6?

I'm going to merge this for now, and if Adam disagrees we can revert. It's not a big deal either way.